_data.py

```diff
@@ -1132,8 +1133,11 @@ def set_data(self, xdata=[1,2,3,4,5], ydata=[[1,2,1,2,1],[3,2,3,4,3]], eydata=No
 
         # plot if not in silent mode
         if self['autoplot']: self.plot()
-
-        return self
+        
+        if self['silent'][0] == True: return None
```

Comment:
what about 
```Python
self._settings['silent'] == True
```

or possibly even a method called ```should_display_text_information()``` which does the silent lookup under a rmore transparent name